test/app/core/managers/audio_record_services_test.dart

```diff
@@ -0,0 +1,313 @@
+      () async {
+        // arrange
+        final mockPermissionHandlerPlatform =
+            PermissionHandlerPlatform.instance;
+        debugDefaultTargetPlatformOverride = TargetPlatform.iOS;
+
+        when((() => mockPermissionHandlerPlatform
+                .checkPermissionStatus(Permission.microphone)))
+            .thenAnswer((_) async => PermissionStatus.granted);
+
+        // act
+        final result = await recordServices.requestPermission();
+
+        // assert
+        expect(result, AudioPermissionState.granted());
+      },
+    );
+
+    test(
+      'requestPermission return a AudioPermissionState.restricted for PermissionStatus.restricted',
+      () async {
+        // arrange
+        final mockPermissionHandlerPlatform =
+            PermissionHandlerPlatform.instance;
+        debugDefaultTargetPlatformOverride = TargetPlatform.iOS;
+
+        when((() => mockPermissionHandlerPlatform
+                .checkPermissionStatus(Permission.microphone)))
+            .thenAnswer((_) async => PermissionStatus.restricted);
+
+        // act
+        final result = await recordServices.requestPermission();
+
+        // assert
+        expect(result, AudioPermissionState.restricted());
+      },
+    );
+
+    test(
+      'dispose() release the audio record session',
+      () async {
+        when(() => soundRecorder.isStopped).thenReturn(false);
+        when(() => audioSyncManager.syncAudio())..thenAnswer((_) async => true);
+        when(() => soundRecorder.stopRecorder()).thenAnswer((_) async => null);
+        when(() => soundRecorder.closeRecorder()).thenAnswer((_) async => null);
+
+        // act
+        await recordServices.dispose();
+
+        // assert
+        verify(() => soundRecorder.isStopped).called(1);
+        verify(() => soundRecorder.stopRecorder()).called(1);
+        verify(() => soundRecorder.closeRecorder()).called(1);
+        verify(() => audioSyncManager.syncAudio()).called(1);
+      },
+    );
+
+    test(
+      'start() record audio for microphone permission granted',
+      () async {
+        // arrange
+        final mockPermissionHandlerPlatform =
+            PermissionHandlerPlatform.instance;
+        debugDefaultTargetPlatformOverride = TargetPlatform.iOS;
+
+        when((() => mockPermissionHandlerPlatform
+                .checkPermissionStatus(Permission.microphone)))
+            .thenAnswer((_) async => PermissionStatus.granted);
+
+        when(() => soundRecorder.isStopped).thenReturn(true);
+        when(() => soundRecorder.openRecorder()).thenAnswer((_) async => null);
+        when(() => soundRecorder.setSubscriptionDuration(any()))
+            .thenAnswer((_) async => null);
+        when(() => soundRecorder.startRecorder(
+              codec: any(named: 'codec'),
+              toFile: any(named: 'toFile'),
+              bitRate: any(named: 'bitRate'),
+              sampleRate: any(named: 'sampleRate'),
+            )).thenAnswer((_) async => null);
```

Comment:
O retorno dos métodos `flutterSoundPlayer.openPlayer()`, `flutterSoundPlayer.setSubscriptionDuration()` e `flutterSoundPlayer.startPlayer()` é `Future<void>`, nunca retornará nulo, acha que faria sentido o mock ser `void` também?
```suggestion
        when(() => soundRecorder.openRecorder()).thenAnswer((_) async {});
        when(() => soundRecorder.setSubscriptionDuration(any()))
            .thenAnswer((_) async {});
        when(() => soundRecorder.startRecorder(
              codec: any(named: 'codec'),
              toFile: any(named: 'toFile'),
              bitRate: any(named: 'bitRate'),
              sampleRate: any(named: 'sampleRate'),
            )).thenAnswer((_) async {});
```